packages/jest-types/__typetests__/expect.test.ts

```diff
@@ -349,27 +351,63 @@ expectError(expect(jest.fn()).toThrowErrorMatchingInlineSnapshot(true));
 
 // extend
 
+type MatcherUtils = typeof jestMatcherUtils & {
+  iterableEquality: Tester;
+  subsetEquality: Tester;
+};
+
 expectType<void>(
   expect.extend({
-    toBeDivisibleBy(actual: number, expected: number) {
+    toBeWithinRange(actual: number, floor: number, ceiling: number) {
+      expectType<number>(this.assertionCalls);
+      expectType<string | undefined>(this.currentTestName);
+      expectType<(() => void) | undefined>(this.dontThrow);
+      expectType<Error | undefined>(this.error);
+      expectType<EqualsFunction>(this.equals);
+      expectType<boolean | undefined>(this.expand);
+      expectType<number | null | undefined>(this.expectedAssertionsNumber);
+      expectType<Error | undefined>(this.expectedAssertionsNumberError);
+      expectType<boolean | undefined>(this.isExpectingAssertions);
+      expectType<Error | undefined>(this.isExpectingAssertionsError);
       expectType<boolean>(this.isNot);
-
-      const pass = actual % expected === 0;
-      const message = pass
-        ? () =>
-            `expected ${this.utils.printReceived(
-              actual,
-            )} not to be divisible by ${expected}`
-        : () =>
-            `expected ${this.utils.printReceived(
-              actual,
-            )} to be divisible by ${expected}`;
-
-      return {message, pass};
+      expectType<string>(this.promise);
+      expectType<Array<Error>>(this.suppressedErrors);
+      expectType<string | undefined>(this.testPath);
+      expectType<MatcherUtils>(this.utils);
+
+      const pass = actual >= floor && actual <= ceiling;
+      if (pass) {
+        return {
+          message: () =>
+            `expected ${actual} not to be within range ${floor} - ${ceiling}`,
+          pass: true,
+        };
+      } else {
+        return {
+          message: () =>
+            `expected ${actual} to be within range ${floor} - ${ceiling}`,
+          pass: false,
+        };
+      }
     },
   }),
 );
 
-// TODO
-// expect(4).toBeDivisibleBy(2);
-// expect.toBeDivisibleBy(2);
+declare module 'expect' {
+  interface AsymmetricMatchers {
+    toBeWithinRange(floor: number, ceiling: number): void;
+  }
+  interface Matchers<R> {
+    toBeWithinRange(floor: number, ceiling: number): R;
+  }
+}
+
+expectType<void>(expect(100).toBeWithinRange(90, 110));
+expectType<void>(expect(101).not.toBeWithinRange(0, 100));
+
+expectType<void>(
+  expect({apples: 6, bananas: 3}).toEqual({
+    apples: expect.toBeWithinRange(1, 10),
+    bananas: expect.not.toBeWithinRange(11, 20),
+  }),
+);
```

Comment:
In this file `expect` is imported from `'@jest/globals'`, but interfaces are extended from `'expect'` module. Nice!

Reply:
oooh, nice! this is perfect, exactly what I was hoping for